products/BellHybrid/apps/application-bell-relaxation/model/RelaxationSongsRepository.cpp

```diff
@@ -0,0 +1,137 @@
+// Copyright (c) 2017-2023, Mudita Sp. z.o.o. All rights reserved.
+// For licensing, see https://github.com/mudita/MuditaOS/LICENSE.md
+#include "RelaxationSongsRepository.hpp"
+#include <module-db/queries/multimedia_files/QueryMultimediaFilesGetLimited.hpp>
+#include <module-db/queries/multimedia_files/QueryMultimediaFilesCount.hpp>
+#include <AsyncTask.hpp>
+#include <vector>
+
+namespace
+{
+    std::uint32_t calculateOffset(const std::uint32_t offset, const std::uint32_t filesCount)
+    {
+        if (offset >= filesCount) {
+            return offset - filesCount;
+        }
+        return offset;
+    }
+    std::pair<std::uint32_t, std::uint32_t> calculateNewOffsetAndLimit(const std::uint32_t offset,
+                                                                       const std::uint32_t limit,
+                                                                       const std::uint32_t loadedFiles)
+    {
+        const auto newLimit  = limit - loadedFiles;
+        const auto newOffset = offset + loadedFiles;
+        return {newOffset, newLimit};
+    }
+} // namespace
+namespace app::relaxation
+{
+
+    RelaxationSongsRepository::RelaxationSongsRepository(ApplicationCommon *application,
+                                                         std::unique_ptr<app::music::AbstractTagsFetcher> tagsFetcher,
+                                                         const std::map<MusicType, std::string> &pathPrefixes)
+        : app::AsyncCallbackReceiver{application}, application{application},
+          tagsFetcher(std::move(tagsFetcher)), pathPrefixes{pathPrefixes}
+    {}
+
+    void RelaxationSongsRepository::init()
+    {
+        for (const auto &p : pathPrefixes) {
```

Comment:
You can use `const auto &[musicType, musicPath] : pathPrefixies`. It should be more readable in the next line `updateFilesCount(musicType, musicPath);`

Reply:
Good point, I missed this :D